### ISSUE TITLE
Deployment demos?

### ISSUE THREAD

**Issue by amellnik** [ndortega/Oxygen.jl#75]
Would it be helpful to include demos or documentation on how to deploy Oxygen APIs to AWS, GCP or similar?  I had a similar project that I just deprecated and I remember that moving from a version running locally to a deployed instance was a common pain point.  If so I'd be glad to add them.

**Comment by ndortega** [comment]
@amellnik 
Absolutely, that's a great idea. In retrospect, that's a pretty big piece of documentation that's missing from the project. I appreciate the help!

**Comment by pboes** [comment]
Absolutely second that. As a non-architect this is what keeps me from using Oxygen for my use case. If there was something like 

https://genieframework.com/docs/genie/v5/tutorials/16--Using_Genie_With_Docker.html

(esp the precompilation bit) and/or analogous to

https://fastapi.tiangolo.com/deployment/deta/

for some service that would be so valuable to me. Right now it boils down a lot of trial and error based off of myriads of bits.

**Comment by damjanmk** [comment]
There is this using Azure: https://discourse.julialang.org/t/tutorial-deploying-an-oxygen-jl-app-to-microsoft-azure/107595

**Comment by arlowhite** [comment]
We're deploying Julia docker containers to [AWS ECS (aka Fargate) ](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html)

These are some of the key issues when you deploy:
* minimize container startup time (compile to CPU targets)
* create a `/health` endpoint. (load balancers pings this to see if container is alive)  
  This may require a prioritized route feature in Oxygen.jl (TODO post issue)
* run with `--threads=auto,1` (important to have a dedicated interactive thread)

This is still a work in progress, we're at the dev stage now, but should be in production within a year. Here's our Dockerfile and ECS config within the CDK project:
* https://github.com/open-AIMS/ReefGuideAPI.jl/blob/main/Dockerfile
* https://github.com/open-AIMS/reefguide-web-api/blob/main/src/infra/components/reefGuideAPI.ts
